### PR TITLE
V1 Design | Header DAO Search

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
Adds DAO search to header

## Notes
Took some liberties since the flow hasn't been designed

## Testing
Enter a valid DAO address and should be able to navigate via a button click

## Screenshots (if applicable)
image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "private": true,
   "dependencies": {
     "@chakra-ui/react": "^2.3.4",
-    "@decent-org/fractal-ui": "^0.0.17",
```
Comment (Da-Colon) [inline package.json:6]: v++ Some input related fixes

File: src/components/menus/AccountDisplay/MenuItems.tsx [hunk 2]
```diff
@@ -17,8 +17,8 @@ export function MenuItems() {
       rounded="lg"
       shadow={'0px 0px 48px rgba(250, 189, 46, 0.48)'}
       mr={['auto', '1rem']}
-      className="menu-list"
       bg="grayscale.black"
+      border="none"
```
Comment (Da-Colon) [inline src/components/menus/AccountDisplay/MenuItems.tsx:21]: Noticed a border on the MenuLIst Element

File: src/components/menus/DAOSearch/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  const { t } = useTranslation(['dashboard']);
+
+  const { errorMessage, loading, address, addressNodeType, resetErrorState, updateSearchString } =
+    useSearchDao();
+
+  const focusInput = () => {
+    if (inputRef.current) {
+      inputRef.current.select();
+    }
+  };
```
Comment (Da-Colon) [inline src/components/menus/DAOSearch/index.tsx:22]: This selects all text onFocus. Pretty handy for this use case

**Comment by herbig** [comment]
@Da-Colon can we also delete the old `DAOSearch` page and commented route in this?

**Comment by Da-Colon** [comment]
> @Da-Colon can we also delete the old `DAOSearch` page and commented route in this?

@herbig I will do a follow up clean up PR as I will remove the link on the splash screen

**Review by herbig — APPROVED** [review]

**Review by herbig — CHANGES_REQUESTED** [review]
@Da-Colon this looks great, one thing, can we clear out any messages if the field is empty?

I don't want to have the user get the error message immediately when they click into the field, also a successful search stays showing if you delete the address from the field

For example, I can search 0xF81246Fd85722Fbc89CE2f48a08c5eACc86dB70a, delete it, then leave the field and click back in and it says DAO found

**Comment by tbwebb22** [comment]
Now, when starting from the "Welcome to Fractal" home page, if I click on "Find a Fractal", it navigates to a blank page. Is that what we want?

**Comment by Da-Colon** [comment]
> @Da-Colon this looks great, one thing, can we clear out any messages if the field is empty?
> 
> I don't want to have the user get the error message immediately when they click into the field, also a successful search stays showing if you delete the address from the field
> 
> For example, I can search 0xF81246Fd85722Fbc89CE2f48a08c5eACc86dB70a, delete it, then leave the field and click back in and it says DAO found

Done.

**Comment by Da-Colon** [comment]
> Now, when starting from the "Welcome to Fractal" home page, if I click on "Find a Fractal", it navigates to a blank page. Is that what we want?

No sir. This will be removed, when I have screens for the splash screen (soon)

**Review by herbig — APPROVED** [review]

**Comment by Da-Colon** [comment]
> Now, when starting from the "Welcome to Fractal" home page, if I click on "Find a Fractal", it navigates to a blank page. Is that what we want?

Since the content-breakpoint pr is really small I removed the link there :D

**Comment by tbwebb22** [comment]
> 

Sweet!

**Review by tbwebb22 — COMMENTED** [review]
Also, when I search for a DAO, and it is found, I click "View DAO", this popup should probably disappear right?

**Comment by Da-Colon** [comment]
> Also, when I search for a DAO, and it is found, I click "View DAO", this popup should probably disappear right?

Ugh, that was a little bit of a challenge, but challenge accepted and completed sir

**Pull request merged**